sql/src/test/java/io/crate/analyze/GroupByAnalyzerTest.java

```diff
@@ -0,0 +1,510 @@
+public class GroupByAnalyzerTest extends CrateDummyClusterServiceUnitTest {
+
+    private SQLExecutor sqlExecutor;
+
+    @Before
+    public void prepare() {
+        DocTableInfo fooUserTableInfo = TestingTableInfo.builder(new TableIdent("foo", "users"), SHARD_ROUTING)
+            .add("id", DataTypes.LONG, null)
+            .add("name", DataTypes.STRING, null)
+            .add("age", DataTypes.INTEGER, null)
+            .add("height", DataTypes.INTEGER, null)
+            .addPrimaryKey("id")
+            .build();
+        DocTableInfoFactory fooTableFactory = new TestingDocTableInfoFactory(
+            ImmutableMap.of(fooUserTableInfo.ident(), fooUserTableInfo));
+        Functions functions = getFunctions();
+        UserDefinedFunctionService udfService = new UserDefinedFunctionService(clusterService, functions);
+        sqlExecutor = SQLExecutor.builder(clusterService)
+            .enableDefaultTables()
+            .addSchema(new DocSchemaInfo("foo", clusterService, functions, udfService, fooTableFactory))
+            .build();
+    }
+
+    private SelectAnalyzedStatement analyze(String statement) {
+        return sqlExecutor.analyze(statement);
+    }
+
+    @Test
+    public void testGroupedSelectMissingOutput() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("column 'load['5']' must appear in the GROUP BY clause or be used in an aggregation function");
+        analyze("select load['5'] from sys.nodes group by load['1']");
+    }
+
+    public void testGroupKeyNotInResultColumnList() throws Exception {
+        SelectAnalyzedStatement analysis = analyze("select count(*) from sys.nodes group by name");
+        assertThat(analysis.relation().querySpec().groupBy().get().size(), is(1));
+        assertThat(analysis.relation().fields().get(0).path().outputName(), is("count(*)"));
+    }
+
+    @Test
+    public void testGroupByOnAlias() throws Exception {
+        QueriedRelation relation = analyze("select count(*), name as n from sys.nodes group by n").relation();
+        assertThat(relation.querySpec().groupBy().get().size(), is(1));
+        assertThat(relation.fields().get(0).path().outputName(), is("count(*)"));
+        assertThat(relation.fields().get(1).path().outputName(), is("n"));
+
+        assertEquals(relation.querySpec().groupBy().get().get(0), relation.querySpec().outputs().get(1));
+    }
+
+    @Test
+    public void testGroupByOnOrdinal() throws Exception {
+        // just like in postgres access by ordinal starts with 1
+        QueriedRelation relation = analyze("select count(*), name as n from sys.nodes group by 2").relation();
+        assertThat(relation.querySpec().groupBy().get().size(), is(1));
+        assertEquals(relation.querySpec().groupBy().get().get(0), relation.querySpec().outputs().get(1));
+    }
+
+    @Test
+    public void testGroupByOnInvalidOrdinal() throws Exception {
```

Comment:
can we move that down to the other `groupByInvalidLiteral` tests and rename it to `testGroupByOnInvalidNegateLiteral` ?